src/BTVNanoCommissioning/workflows/BTA_ttbar_producer.py

```diff
@@ -515,7 +524,10 @@ def process_shift(self, events, shift_name):
         passJetSel = (
             ((abs(chsel) == 11) | (abs(chsel) == 13)) & (ak.num(Jet_clean) >= 4)
         ) | ((abs(chsel) > 13) & (ak.num(Jet_clean) >= 1))
-        passMetSel = events.PuppiMET.pt > 0
+        if "Run3" in self._campaign:
+            passMetSel = events.PuppiMET.pt > 0
+        else:
+            passMetSel = events.MET.pt > 0
```

Comment:
Apologies, did not find out the implementation was using the JME_shifts
Indeed in the functions we already implement the JEC with [puppiMET](https://github.com/cms-btv-pog/BTVNanoCommissioning/blob/master/src/BTVNanoCommissioning/utils/correction.py#L554) and replace it as MET branch.

If you use the puppiMET branch, this might not propagate the JEC correction correctly

Reply:
Hi @philippgadow could you please fix this and I think we are good to merge :) thanks!

Reply:
sorry for the delay, it is fixed in a more recent commit, now it says

```
basic_vars["ttbar_met_pt"] = events.MET.pt
basic_vars["ttbar_met_phi"] = events.MET.phi
```
without the if/else